src/middleware/apiKeyChecker.ts

```diff
@@ -0,0 +1,29 @@
+import fs from 'fs';
+import { AuthenticationError } from '../errors';
+import type {
+  NextFunction,
+  Request,
+  Response,
+} from 'express';
+
+export const checkApiKey = (req: Request, res: Response, next: NextFunction): void => {
+  const apiKey = req.headers['x-api-key'] ?? '';
+  if (apiKey === '') {
+    next(new AuthenticationError(
+      'API key not provided in the header',
```

Comment:
Add the exact header expected in the message, e.g. 'API key not provided in header "x-api-key"'.